gratipay/models/package/__init__.py

```diff
@@ -40,3 +51,35 @@ def from_names(cls, package_manager, name):
         """
         return cls.db.one("SELECT packages.*::packages FROM packages "
                           "WHERE package_manager=%s and name=%s", (package_manager, name))
+
+
+    @property
+    def team(self):
+        """A computed attribute, the :py:class:`~gratipay.models.team.Team`
+        linked to this package if there is one, otherwise ``None``. Makes a
+        database call.
+        """
+        return self._load_team(self.db)
+
+
+    def ensure_team(self, cursor, owner):
+        """Given a db cursor and :py:class:`Participant`, insert into ``teams`` if need be.
+        """
+        team = self._load_team(cursor)
```

Comment:
I captured the [SQL](https://gist.github.com/whit537/86175f87a7003b3366c2ff1c15ac3b8c) for a single `finish_email_verification` call. The best option I see is adding `package_id` to `teams` so that committing subsequent transactions will fail with `IntegrityError` (I think?). Currently we have `packages.team_id` so we'd need to decide whether to have the link both ways or replace the one with the other.

The alternative I see is to use some sort of explicit Postgres locking like we do in [`cron.py`](https://github.com/gratipay/gratipay.com/blob/8da4f9cbcd38ad06514a5b805f0579e528b07084/gratipay/cron.py#L42). That seems less natural to me but if the other idea doesn't work out for some reason maybe we have to go this route.

This also raises the question of where all in the app we have concurrency bugs: audit reticketed as https://github.com/gratipay/gratipay.com/issues/4418.